cpp/src/join/join_common_utils.cuh

```diff
@@ -158,24 +154,29 @@ get_trivial_left_join_indices(
  *
  */
 template <typename MultimapType>
-void build_join_hash_table(cudf::table_view const& build,
-                           MultimapType& hash_table,
-                           null_equality const nulls_equal,
-                           [[maybe_unused]] bitmask_type const* bitmask,
-                           rmm::cuda_stream_view stream)
+void build_join_hash_table(
+  cudf::table_view const& build,
+  MultimapType& hash_table,
+  null_equality const nulls_equal,
+  [[maybe_unused]] bitmask_type const* bitmask,
+  rmm::cuda_stream_view stream,
+  std::shared_ptr<experimental::row::equality::preprocessed_table> preprocessed_build = nullptr)
```

Comment:
I will remove the default parameter and re-order it when I raise a PR for updated `mixed_join_*` APIs to experimental row operators